README.md

```diff
@@ -50,6 +50,6 @@ reasoning, to this markdown file.
 
 - Worst Case Memory Complexity : $O(n * 2^n)$
 * the cache stores $2^n$ keys every call, therefore the cache will store $n * 2^n$ elements when the algorithm is done running since we may have multiple of the same permutations that are unique because they will have different start values.  
-- Worst Case Time Complexity : $O(n * 2^n)$
-* For each cache entry we create a 'newcities' which takes time complexity of $O(n)$. We also run a 'for' loop 'n' times to find the next starting point, and the new 'newcities' array. Therefore for each cache entry we must have time complexity of $O(n + n)$. It is worth noting that our for loop inside the tsp function will not necessarly run 'n' times where n is the size of the input matrix, but where 'n' is the size of the 'nextcites' array, which will always be less than or equal to the size of the input array. Since we reduce the number of subproblems stored in the cache by using memoization, the number of calls here would not be n! but instead would be 2^n. And since we need to check all start options, in the worst case, our time complexity will end up being $O((n + n) * (n * 2^n))$ which can be simplified to $O(n * 2^n)$. 
+- Worst Case Time Complexity : $O(n^2 * 2^n)$
+* For cache each entry we have a constant amount of work where we run a 'for' loop 'n-1' times (until the array is of size 1) to find the next starting point, and the new 'newcities' array (which takes a time complexity of $O(n)$). Since we reduce the number of subproblems stored in the cache by using memoization, the number of calls here would not be n! but instead would be 2^n. And since we need to check all start options, in the worst case, our time complexity will end up being $O((n * n) * 2^n))$ which can be simplified to $O(n^2 * 2^n)$. 
```

Comment:
Not following you again. You say that there's a constant amount of work, but in the same sentence that a loop runs a number of times proportional to the input size.